security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysis.java

```diff
@@ -89,12 +111,85 @@ protected SecurityAnalysisResult readResults(Path workingDir) {
+
+        private SubTaskHandler simpleSubTaskHandler;
+
+        private static final String SA_LOG_TAR = "sa_logs.tar";
+        private static final String SA_LOG_TAR_GZ = SA_LOG_TAR + ".gz";
+
+        public SubTaskWithLogHandler(SubTaskHandler subTaskHandler) {
+            this.simpleSubTaskHandler = Objects.requireNonNull(subTaskHandler);
+        }
+
+        @Override
+        public List<CommandExecution> before(Path workingDir) throws IOException {
+            simpleSubTaskHandler.before(workingDir); // invocation just for prepare input files
+            // override cmd with log
+            SimpleCommand cmd = simpleSubTaskHandler.baseCmdBuilder()
+                    .option(OUTPUT_LOG_OPTION, this::getLogOutputName)
                     .build();
-            return Collections.singletonList(new CommandExecution(cmd, actualTaskCount, 1));
+            return Collections.singletonList(new CommandExecution(cmd, simpleSubTaskHandler.actualTaskCount, 1));
+        }
+
+        private String getLogOutputName(int taskNumber) {
+            return "logs_" + taskNumber + ".xml.gz";
+        }
+
+        @Override
+        public SecurityAnalysisResultWithLog after(Path workingDir, ExecutionReport report) throws IOException {
+            SecurityAnalysisResult re = simpleSubTaskHandler.after(workingDir, report);
+            archiveLogs(workingDir);
+            compressLogs(workingDir);
+            Files.delete(workingDir.resolve(SA_LOG_TAR));
+            return new SecurityAnalysisResultWithLog(re, workingDir.resolve(SA_LOG_TAR_GZ).toFile());
+        }
+
+        private void archiveLogs(Path workingDir) throws IOException {
+            try (TarArchiveOutputStream taos = new TarArchiveOutputStream(new FileOutputStream(workingDir.resolve(SA_LOG_TAR).toFile()))) {
+                for (int i = 0; i < simpleSubTaskHandler.actualTaskCount; i++) {
+                    String hadesLogName = getLogOutputName(i);
+                    addFile(taos, workingDir, hadesLogName);
+                    addFile(taos, workingDir, SubTaskHandler.SA_TASK_CMD_ID + "_" + i + ".out");
+                    addFile(taos, workingDir, SubTaskHandler.SA_TASK_CMD_ID + "_" + i + ".err");
+                }
+                taos.flush();
+            }
+        }
+
+        private void addFile(TarArchiveOutputStream taos, Path workingDir, String fileName) throws IOException {
+            Path path = workingDir.resolve(fileName);
+            ArchiveEntry archiveEntry = taos.createArchiveEntry(path.toFile(), fileName);
+            taos.putArchiveEntry(archiveEntry);
+            try (InputStream is = Files.newInputStream(path)) {
+                IOUtils.copy(is, taos);
+            }
+            taos.closeArchiveEntry();
+        }
+
+        private void compressLogs(Path workingDir) throws IOException {
+            try (InputStream in = Files.newInputStream(workingDir.resolve(SA_LOG_TAR));
```

Comment:
Does this have an added value instead of just using for example IOUtils.copy(inputStream, new Gzip***) ?